connectors/byoc.py

```diff
@@ -554,3 +554,77 @@ async def sync(self, elastic_server, idling, sync_now=False):
         finally:
             self._syncing = False
             self._start_time = None
+
+STUCK_JOBS_THRESHOLD = 60 # 60 seconds
+
+class SyncJobIndex(ESIndex):
+    """
+    Represents Elasticsearch index for sync jobs
+
+    Args:
+        index_name (str): Name of an Elasticsearch index
+        elastic_config (dict): Elasticsearch configuration and credentials
+    """
+    def __init__(self, index_name, elastic_config):
+        super().__init__(index_name=index_name, elastic_config=elastic_config)
+
+    def _create_object(self, doc_source):
+        """
+        Args:
+            doc_source (dict): A raw Elasticsearch document
+        Returns:
+            SyncJob
+        """
+        return SyncJob(
+            self,
+            connector_id=doc_source['_source']["connector"]["id"],
+            doc_source=doc_source['_source'],
+        )
+
+    def pending_job_query(self, connectors_ids=[]):
```

Comment:
If you are referring to this [code](https://github.com/elastic/connectors-ruby/blob/main/lib/core/connector_job.rb#L54) then my answer is No.  I want to keep SyncJobIndex simple and contain only sync jobs-related logic.

Reply:
> If you are referring to this [code](https://github.com/elastic/connectors-ruby/blob/main/lib/core/connector_job.rb#L54) then my answer is No.

The logic is a bit different from ruby. Python will just pick up jobs of `native_connector_types` and `connector_id` configured in yaml. 

>I want to keep SyncJobIndex simple and contain only sync jobs-related logic.

I think it's still sync jobs-related logic by changing it to 
```
def pending_job_query(self, native_connector_types=[], connectors_ids=[]):
```

Otherwise, when the consumer tries pull jobs, it has to query connectors first, then jobs.

Reply:
To get the `connectors_ids` for the specific connector types you still need to perform a query right? AFAIK we don't store any information about connectors type in the sync jobs index. So how would you use `native_connector_types ` then?

Reply:
>AFAIK we don't store any information about connectors type in the sync jobs index.

We did so in [ruby](https://github.com/elastic/connectors-ruby/blob/a86347622b6947bbfc3b43bde00f5fca13789e29/lib/core/elastic_connector_actions.rb#L200), and will do so soon in python.

But I found that we [query twice](https://github.com/elastic/connectors-ruby/blob/a86347622b6947bbfc3b43bde00f5fca13789e29/lib/core/jobs/consumer.rb#L85-L92) in Ruby as well, where I do think we can improve.

Reply:
Would it work for you if we do this improvement later when we the service actually starts storing `native_connector_types`?

Reply:
Let's proceed first, not a blocker